packages/lesswrong/components/sunshineDashboard/ModeratorActions.tsx

```diff
@@ -293,59 +296,39 @@ export const ModeratorActions = ({classes, user, currentUser, refetch, comments,
     setNotes( newNotes )
   }
 
-  const mostRecentRateLimit = user.moderatorActions.find(modAction => modAction.type === RATE_LIMIT_ONE_PER_DAY);
 
-  const createRateLimit = async (endDate?: Date) => {
-    const newNotes = getModSignatureWithNote(`rate limit added`) + notes;
-    void updateUser({
-      selector: { _id: user._id },
-      data: { sunshineNotes: newNotes }
-    });
+  const createRateLimit = async (type: RateLimitType) => {
 
-    const maybeEndedAt = endDate ? { endedAt: endDate } : {};
-    // Otherwise, we want to create a new one
+    const endDate = new Date();
+    endDate.setDate(endDate.getDate() + 60);
+    
     await createModeratorAction({
       data: {
-        type: RATE_LIMIT_ONE_PER_DAY,
+        type,
         userId: user._id,
-        ...maybeEndedAt
+        endedAt: endDate
       }
     });
 
-    setNotes(newNotes);
+    const existingRateLimits = user.moderatorActions.filter(modAction => rateLimits.includes(modAction.type as RateLimitType)) ?? [];
```

Comment:
`rateLimits.includes(modAction.type as RateLimitType)` seems like a good place to instead use the new `TupleSet` functionality, which should let you avoid the type-cast (by creating a TupleSet out of `rateLimits`, and then just using the resulting `.has` type guard).